clients/spark/core/src/main/scala/io/treeverse/clients/StorageClients.scala

```diff
@@ -125,14 +77,18 @@ object StorageClients {
     }
   }
 
-  def apply(storageType: String, configMapper: ConfigMapper): StorageClient = {
+  def apply(
+      storageType: String,
+      configMapper: ConfigMapper,
+      storageNamespace: String,
+      region: String
+  ): StorageClient = {
     storageType match {
       case StorageTypeS3 =>
-        new S3(configMapper)
+        new S3(storageNamespace, region, S3.S3NumRetries, configMapper)
```

Comment:
Prefer to have configMapper either always first or always last.  Here it seems opposite to l. 90.